### PR TITLE
Replace the date_created column name with a class property in the analytics DataStore

### LINKED ISSUE #32643: Replace the `date_created` column name with a class property in the analytics DataStore
Hey team!

While working on extending several extensions into the WCA Analytics, we encountered an extensibility issue regarding the date's table column name.

More precisely, under the `Automattic\WooCommerce\Admin\API\Reports\DataStore` class, the `date_created` DB column appears hard-coded in a few places. 

See: https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Admin/API/Reports/DataStore.php#L502

And in the `Automattic\WooCommerce\Admin\API\Reports\TimeInterval` class: https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Admin/API/Reports/TimeInterval.php#L84

This leads to difficulties when we need to extend analytics using custom DB tables rather than the lookup tables.

### Describe the solution

Introduce a class property `$date_column_name` with default value `(= date_created)` and replace every occurrence with this protected prop.

Also, pass an additional argument into the `TimeInterval:: db_datetime_format` to specify the date column name.

### Additional context

Currently, the Gift Cards extension extends the base DataStore class to replace all the methods that include the hardcoded values.

### PR DESCRIPTION
**Pull request by xristos3490**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Please see the issue for the description.

Closes #32643

### How to test the changes in this Pull Request:

N/A

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?
* [ ] Have you created a changelog file by running `pnpm nx affected --target=changelog`?

### FOR PR REVIEWER ONLY:

* [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce/src/Admin/API/Reports/TimeInterval.php [hunk 1]
```diff
@@ -79,28 +79,29 @@ public static function default_after() {
 	 *
 	 * @param string $time_interval Time interval.
 	 * @param string $table_name Name of the db table relevant for the date constraint.
+	 * @param string $date_column_name Name of the date table column.
 	 * @return mixed
 	 */
-	public static function db_datetime_format( $time_interval, $table_name ) {
+	public static function db_datetime_format( $time_interval, $table_name, $date_column_name ) {
```
Comment (chihsuan) [inline plugins/woocommerce/src/Admin/API/Reports/TimeInterval.php]:
```suggestion
	public static function db_datetime_format( $time_interval, $table_name, $date_column_name = 'date_created' ) {
```

Could we add a default value for `$date_column_name` for backward compatibility?

**Review by chihsuan — CHANGES_REQUESTED** [review]
@xristos3490 Thanks for working on this! I just left a comment and it would be great if you could add a changelog entry via `pnpm nx  changelog woocommerce`

File: plugins/woocommerce/src/Admin/API/Reports/TimeInterval.php (hunk 1 above)
Comment (xristos3490) [reply plugins/woocommerce/src/Admin/API/Reports/TimeInterval.php]: Great catch, @chihsuan! 🎉

**Comment by xristos3490** [comment]
Thanks for the review, @chihsuan! Changelog file added 👍

**Comment by botwoo** [comment]
:bar_chart: Test reports for this pull request have been published and are accessible through the following links:

- [REST API Test Report](https://woocommerce.github.io/woocommerce-test-reports/pr/32644/api/)
- E2E Test Report (TBC)

Latest commit referenced in the reports: `Add changelog` 58abc685cbe446368436dd64ed2ca594d3d45f90
This comment will automatically be updated with the latest referenced commit when you push new changes to this pull request.

---

_Visit the [WooCommerce Test Reports](https://woocommerce.github.io/woocommerce-test-reports) homepage to view all published reports. See the [FAQs](https://github.com/woocommerce/woocommerce-test-reports/wiki/FAQs) page if you're having problems accessing them._

**Review by chihsuan — APPROVED** [review]
Thanks @xristos3490. Looks good! 🚢

**Pull request merged**